lib/src/digest_sink.dart

```diff
@@ -7,24 +7,18 @@ import 'digest.dart';
 /// A sink used to get a digest value out of `Hash.startChunkedConversion`.
 class DigestSink extends Sink<Digest> {
   /// The value added to the sink, if any.
-  Digest get value {
-    assert(_value != null);
-    return _value;
-  }
+  Digest get value => _value;
 
-  Digest _value;
+  late Digest _value;
 
   /// Adds [value] to the sink.
   ///
   /// Unlike most sinks, this may only be called once.
   @override
   void add(Digest value) {
-    assert(_value == null);
     _value = value;
   }
 
   @override
-  void close() {
-    assert(_value != null);
-  }
+  void close() {}
```

Comment:
I think this could be removed instead

Reply:
cc @jakemac53 - are you concerned at that we're losing this assert, or that we're turning debug only failures into always failures? I think it's probably OK but I'm curious what you think.

Reply:
seems like it IS needed – it's abstract.

Reply:
We could consider a read of the variable to regain the check that it was assigned at some point. We could even do it in an `assert` if we wanted. I don't think we can replace the other one though.

```dart
void close() {
  assert(() {
    _value;
    return true;
  });
}
```

@leafpetersen - does using a `late final` field as a statement have the effect of ensuring it was assigned?

Reply:
yes.